src/runtime/hexagon_remote/qdebug_dump_reg_asm.S

```diff
@@ -0,0 +1,51 @@
+/**=============================================================================
+
+ @file
+ qdebug_dump_reg_asm.h
+
+ @brief
```

Comment:
was this file copies from somewhere else?

also, we do remove copyrights before upstreaming.

The test function is unused?

perhaps we can rewrite this in CPP using Inline asms?

what is  the minimum level of SDK that libqdebug ships in ? looks like 3.3 to me

i would suggest we have the libqdebug folks add a precompiled dump all VREGS call for us, then we can just call it and not have to build this file, we simply link against libqdebug when we build the librar.y

Reply:
This file was copied from SDK sample for qdebug and modified to dump only register contents.
This function is called in halide_remote.cpp.
I inquired with qdebug team if it can be done with inline asm, they suggested to use the sample assembly file to generate object file and link with our code. In the SDK sample too, it is done in similar way.
Min level of SDK required is 3.3
print functions are available in qdebug library. We can use them in .S files in a function and call those functions from .cpp file.